### PR TITLE
blockdev_inc_backup_merge_with_nonexist_bitmap:Merge with a non-existed bitmap

### PR DESCRIPTION
**Pull request by aliang123**
Merge with a non-existed bitmap

Signed-off-by: Aihua Liang <aliang@redhat.com>
id:1984268

### CODE REVIEW

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_merge_with_nonexist_bitmap.q35: PASS (42.27 s)

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_merge_with_nonexist_bitmap.q35: PASS (42.37 s)

**Comment by aliang123** [comment]
@zhencliu @ngu-niny, please help to review, thanks

File: qemu/tests/blockdev_inc_backup_merge_with_nonexist_bitmap.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        ]
+        try:
+            self.main_vm.monitor.transaction(job_list)
+        except QMPCmdError as e:
+            nonexist_bitmap = self._merged_bitmaps[1]
+            qmp_error_msg = self.params.get("qmp_error_msg") % nonexist_bitmap
+            if qmp_error_msg not in str(e.data):
+                self.test.fail(str(e))
+        else:
+            self.test.fail("Inc backup with invalid bitmap mode:%s"
```
Comment (zhencliu) [inline qemu/tests/blockdev_inc_backup_merge_with_nonexist_bitmap.py:40]: Please update error msg, the current error msg is not for this case

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_merge_with_nonexist_bitmap.q35: PASS (42.68 s)

File: qemu/tests/blockdev_inc_backup_merge_with_nonexist_bitmap.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        try:
+            self.main_vm.monitor.transaction(job_list)
+        except QMPCmdError as e:
+            nonexist_bitmap = self._merged_bitmaps[1]
+            qmp_error_msg = self.params.get("qmp_error_msg") % nonexist_bitmap
+            if qmp_error_msg not in str(e.data):
+                self.test.fail(str(e))
+        else:
+            self.test.fail("Can merge with a non-exist bitmap:%s"
+                           % nonexist_bitmap)
```
Comment (zhencliu) [inline qemu/tests/blockdev_inc_backup_merge_with_nonexist_bitmap.py:41]: nonexist_bitmap -> self._merged_bitmaps[1]
If the program enters into the else block, nonexist_bitmap is a var undefined

File: qemu/tests/blockdev_inc_backup_merge_with_nonexist_bitmap.py (hunk 1 above)
Comment (aliang123) [reply qemu/tests/blockdev_inc_backup_merge_with_nonexist_bitmap.py]: have updated, thanks zhenchao.

File: qemu/tests/blockdev_inc_backup_merge_with_nonexist_bitmap.py (hunk 2 above)
Comment (aliang123) [reply qemu/tests/blockdev_inc_backup_merge_with_nonexist_bitmap.py]: have updated, thanks zhenchao.

**Comment by ngu-niny** [comment]
1-Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_inc_backup_merge_with_nonexist_bitmap Finshed 2021-07-28 04:21:57 2021-07-28 04:24:11 PASS 133.617436

Acked-by: Nini Gu <ngu@redhat.com>

**Comment by zhencliu** [comment]
LGTM

**Pull request merged**